homework_02/task_01.py

```diff
@@ -67,40 +67,40 @@ def get_rarest_char(file_path: str, encoding: str = "unicode-escape") -> str:
     """
     Find rarest symbol for document.
     """
-    chars = [
-        symb
-        for token_type, symb in tokenize(file_path, encoding)
-        if token_type == "symbol"
-    ]
-    char_counter = Counter(chars)
-    n_min = len(chars)
-    for symb, n in char_counter.items():
-        if n < n_min:
-            symb_min = symb
-            n_min = n
-    return symb_min
+    char_counter = Counter(
+        [
+            symb
+            for token_type, symb in tokenize(file_path, encoding)
+            if token_type == "symbol"
+        ]
+    )
```

Comment:
list comprehension -> generator expression.  You can omit parenthesis if you have only one argument.

```suggestion
    char_counter = Counter(
            symb
            for token_type, symb in tokenize(file_path, encoding)
            if token_type == "symbol"
    )
```

Reply:
Thanks!
BTW, in English parentheses are these: `( )`. `[ ]` - these are brackets. ;)

Reply:
parentheses is used for generator expression and you can omit them in some cases.   `sum((1 for _ in "abc"))` - > `sum(1 for _ in "abc")`